pkg/cmd/stack.go

```diff
@@ -154,9 +154,9 @@ func newStackCmd() *cobra.Command {
 	cmd.PersistentFlags().StringVarP(
 		&stackName, "stack", "s", "",
 		"The name of the stack to operate on. Defaults to the current stack")
-	cmd.PersistentFlags().BoolVarP(
+	cmd.Flags().BoolVarP(
 		&showIDs, "show-ids", "i", false, "Display each resource's provider-assigned unique ID")
-	cmd.PersistentFlags().BoolVarP(
+	cmd.Flags().BoolVarP(
 		&showURNs, "show-urns", "u", false, "Display each resource's Pulumi-assigned globally unique URN")
 	cmd.PersistentFlags().BoolVar(
```

Comment:
What about `show-secrets`?

```suggestion
	cmd.Flags().BoolVar(
```

Reply:
I didn't change that as I wasn't sure if it affected `stack output` or not.

Reply:
I actually went back and looked at the reason for this. `--show-ids` and `--show-urns` are flags for `pulumi stack` itself, which `--show-secrets` I believe to affect `export` and such.

Reply:
I looked into this in more detail.

There is a `pulumi stack output` command which also accepts `--show-secrets`, but it redefines this flag for itself.

There's no other commands under `stack` that refers to its `showSecrets` bool variable or is otherwise looking for "show-secrets", so we should be able to change it here from `PersistentFlags` to `Flags`.